File: JavaScript/256-Paint House.js
/**
 * @param {number[][]} costs
 * @return {number}
 */
var minCost = function(costs) {
    const n = costs.length;
    
    for (let i=1; i<n; i++){
        for (let j=0; j<3; j++){
            costs[i][j] += Math.min(costs[i-1][((j-1)+3)%3], costs[i-1][(j+1)%3]);
        }
    }
    
    return Math.min(...costs[costs.length-1]);
};

File: C++/256-Paint House.cpp
class Solution {
public:
    int minCost(vector<vector<int>>& costs) {
        int N = costs.size();
        for (int r=1; r<N; r++) {
            costs[r][0] += min(costs[r-1][1], costs[r-1][2]);
            costs[r][1] += min(costs[r-1][0], costs[r-1][2]);
            costs[r][2] += min(costs[r-1][0], costs[r-1][1]);
        }
        return *min_element(costs[N-1].begin(), costs[N-1].end());
    }
};